services/frontend-service/pkg/interceptors/interceptors.go

```diff
@@ -132,16 +149,44 @@ func DexLoginInterceptor(
 	w http.ResponseWriter,
 	req *http.Request,
 	httpHandler http.HandlerFunc,
-	clientID, baseURL string,
+	clientID, baseURL, DexRbacPolicyPath string,
 ) {
-	role, err := auth.VerifyToken(req.Context(), req, clientID, baseURL)
+	httpCtx, err := GetContextFromDex(w, req, clientID, baseURL, DexRbacPolicyPath)
 	if err != nil {
 		logger.FromContext(req.Context()).Debug(fmt.Sprintf("Error verifying token for Dex: %s", err))
 		// If user is not authenticated redirect to the login page.
 		http.Redirect(w, req, auth.LoginPATH, http.StatusFound)
+		return
 	}
-	auth.WriteUserRoleToHttpHeader(req, role)
-	httpCtx := auth.WriteUserRoleToGrpcContext(req.Context(), role)
 	req = req.WithContext(httpCtx)
 	httpHandler(w, req)
 }
+
+func GetContextFromDex(w http.ResponseWriter, req *http.Request, clientID, baseURL, DexRbacPolicyPath string) (context.Context, error) {
+	claims, err := auth.VerifyToken(req.Context(), req, clientID, baseURL)
+	if err != nil {
+		logger.FromContext(req.Context()).Info(fmt.Sprintf("Error verifying token for Dex: %s", err))
+		return req.Context(), err
+	}
+	httpCtx := req.Context()
+	switch val := claims["groups"].(type) {
+	case []interface{}:
+		for _, group := range val {
+			groupName := strings.Trim(group.(string), "\"")
+			httpCtx = CheckPolicy(httpCtx, w, req, groupName, DexRbacPolicyPath)
+		}
+	case []string:
+		httpCtx = CheckPolicy(httpCtx, w, req, strings.Join(val, ","), DexRbacPolicyPath)
+	case string:
+		httpCtx = CheckPolicy(httpCtx, w, req, val, DexRbacPolicyPath)
+	}
+
+	if claims["email"].(string) != "" {
+		httpCtx = CheckPolicy(httpCtx, w, req, claims["email"].(string), DexRbacPolicyPath)
+	}
```

Comment:
please rephrase this so that we have the structure 
```
if errorCase {
   return err
}
if otherErrCase {
return otherErr
}
return success.
```
Otherwise it's hard to follow what is happening, and it saves you from checking the claims["email"] twice.